agent/api/container/container.go

```diff
@@ -109,15 +106,25 @@ type HealthStatus struct {
 	Output string `json:"output,omitempty"`
 }
 
-type ManagedAgent struct {
+type ManagedAgentState struct {
+	// ID of this managed agent state
+	ID string `json:"id:omitempty"`
```

Comment:
what happens if we don't json save this `ID`?

Reply:
I'm using this `ID` to store the `UUID` that we append to the `/ecs-exec-command-UUID`. We need to persist this since it is generated in `execcmd.InitializeContainer` and used in `execcmd.StartAgent`, we need to remember it even after restarts